torchx/schedulers/kueue_job_scheduler.py

```diff
@@ -0,0 +1,809 @@
+        assert local_kueue is None or isinstance(
+            local_kueue, str
+        ), "local_kueue must be a str"
+        
+        kueue_priority_class = cfg.get("kueue_priority_class")
+        assert kueue_priority_class is None or isinstance(
+            kueue_priority_class, str
+        ), "kueue_priority_class must be a str"
+        
+        resource = app_to_resource(app, service_account, local_kueue,  kueue_priority_class)
+        req = KueueJob(
+            resource=resource,
+            images_to_push=images_to_push,
+        )
+        return AppDryRunInfo(req, repr)
+
+    def _validate(self, app: AppDef, scheduler: str) -> None:
+        # Skip validation step
+        pass
+
+    def _cancel_existing(self, app_id: str) -> None:
+        from kubernetes import client
+        namespace, name = app_id.split(":")
+
+        self._batchv1_api().delete_namespaced_job(
+            namespace=namespace,
+            name=name,
+            body=client.V1DeleteOptions(
+                    propagation_policy='Foreground',
+                    grace_period_seconds=5))
+
+    def _run_opts(self) -> runopts:
+        opts = runopts()
+        opts.add(
+            "namespace",
+            type_=str,
+            help="Kubernetes namespace to schedule Job in",
+            default="default",
+        )
+        opts.add(
+            "service_account",
+            type_=str,
+            help="The service account name to set on the pod specs",
+        )
+        opts.add(
+            "local_kueue",
+            type_=str,
+            help="The Local Kueue name to set on the local Kueue label",
+        )
+        opts.add(
+            "kueue_priority_class",
+            type_=str,
+            help="The kueue priority class name to use for the priority class label",
+        )
+        return opts
+
+    def describe(self, app_id: str) -> Optional[DescribeAppResponse]:
+        from kubernetes import client
+        
+        namespace, name = app_id.split(":")
+        roles = {}
+        roles_statuses = {}
+
+        try:
+            api_instance = self._batchv1_api()
+            job = api_instance.read_namespaced_job_status(name, namespace)
+        except client.ApiException as e:
+            return f"Exception: {e}"
+        
+
+        try: 
+            status = job.status
+        except:
+            status = None
+        
+        if status:
+            condition = status.conditions[0]
+            if condition.type == "Suspended" and condition.reason == "JobSuspended":
+                app_state = JOB_STATE["Suspended"]
+            elif condition.reason == "JobResumed":
+                if status.active is not None:
+                    app_state = JOB_STATE["Running"]
+                elif status.succeeded is not None:
+                    app_state = JOB_STATE["Completed"]
+                elif status.failed is not None:
+                    app_state = JOB_STATE["Failed"]
```

Comment:
From my understanding of https://docs.openshift.com/container-platform/4.14/rest_api/workloads_apis/job-batch-v1.html#status, `Completed`, `Running`, and `Failed` will be integer values representing the number of pods in each state. As it stands, if there are 2 pods that succeed and 2 that fail your app_state would be `Completed`.

Reply:
I see so a job could be considered a failure and a success with the current set up. I will look into this and come up with another way of gauging the status :+1: